internal/elasticsearch/ingest/datastream.go

```diff
@@ -70,6 +89,27 @@ func loadIngestPipelineFiles(dataStreamPath string, nonce int64) ([]Pipeline, er
 			pipelineTag := s[1]
 			return []byte(getPipelineNameWithNonce(pipelineTag, nonce))
 		})
+
+		// Unmarshal the YAML data into a ESIngestPipeline struct
+		var esPipeline ESIngestPipeline
+		err = yaml.Unmarshal(c, &esPipeline)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal ingest pipeline YAML data (path: %s): %w", path, err)
+		}
+
+		// read routing_rules.yml and convert it into reroute processors in ingest pipeline
+		rerouteProcessors, err := loadRoutingRuleFile(dataStreamPath)
```

Comment:
Could you please add a test package that has a routing rule file?

Reply:
Yes thank you!! I added a package called awsfirehose with a routing rule file, how does this test package work?